examples/madry_lab_challenges/madry_mnist_model.py

```diff
@@ -0,0 +1,79 @@
+"""cleverhans.model.Model implementation of mnist_challenge.model.Model
+
+This re-implementation factors variable creation apart from forward
+propagation so it is possible to run forward propagation more than once
+in the same model.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from collections import OrderedDict
+import tensorflow as tf
+from cleverhans.model import Model
+
+
+class MadryMNIST(Model):
+
+    def __init__(self, **kwargs):
+        # NOTE: for compatibility with Madry Lab downloadable checkpoints,
+        # we cannot use scopes, give these variables names, etc.
+        self.W_conv1 = self._weight_variable([5, 5, 1, 32])
+        self.b_conv1 = self._bias_variable([32])
+        self.W_conv2 = self._weight_variable([5, 5, 32, 64])
+        self.b_conv2 = self._bias_variable([64])
+        self.W_fc1 = self._weight_variable([7 * 7 * 64, 1024])
+        self.b_fc1 = self._bias_variable([1024])
+        self.W_fc2 = self._weight_variable([1024, 10])
+        self.b_fc2 = self._bias_variable([10])
+        super(MadryMNIST, self).__init__(**kwargs)
+
+    def fprop(self, x):
+
+        output = OrderedDict()
+        # first convolutional layer
+        h_conv1 = tf.nn.relu(self._conv2d(x, self.W_conv1) + self.b_conv1)
+        h_pool1 = self._max_pool_2x2(h_conv1)
+
+        # second convolutional layer
+        h_conv2 = tf.nn.relu(
+            self._conv2d(h_pool1, self.W_conv2) + self.b_conv2)
+        h_pool2 = self._max_pool_2x2(h_conv2)
+
+        # first fully connected layer
+
+        h_pool2_flat = tf.reshape(h_pool2, [-1, 7 * 7 * 64])
+        h_fc1 = tf.nn.relu(tf.matmul(h_pool2_flat, self.W_fc1) + self.b_fc1)
+
+        # output layer
+        logits = tf.matmul(h_fc1, self.W_fc2) + self.b_fc2
+
+        l = locals()
+        for key in sorted(l.keys()):
+            if key in ["self", "output"]:
+                continue
+            output[key] = l[key]
```

Comment:
`sorted(locals())` actually gives you a list of the sorted keys, so that wouldn't work quite as written. You still need to iterate over the output of `sorted` to package up the values along with the keys. I factored it into a separate function though.